spec/datadog/core/workers/runtime_metrics_spec.rb

```diff
@@ -282,7 +282,11 @@
       it 'produces metrics every interval' do
         worker.perform
 
-        sleep(default_flush_interval + 0.2)
+        Thread.pass # Let the background thread start
+
+        sleep(default_flush_interval * 1.1)
+
+        Thread.pass # Let the background thread run, if it hasn't during the sleep above
```

Comment:
I'm not sure the `Thread.pass` are needed -- Ruby should (will?) give other threads an opportunity to run when the current thread is `sleeping`.

Reply:
The first one, before sleep, is the one I think is important, because it "almost" guarantees that the background thread has had a chance to run and it likely already sleeping on its side. Without it, we normally start counting down the main thread sleep before the background thread. But I removed both `Thread.pass`.